deploy/test-environments/fleet_api/src/api/base_call_api.py

```diff
@@ -0,0 +1,61 @@
+"""
+This module provides functionality for performing API calls and handling API call exceptions.
+It utilizes the 'requests' library for making HTTP requests.
+
+Module contents:
+    - APICallException: Exception class raised for API call failures.
+    - perform_api_call: Function for making API calls and handling the response.
+
+Dependencies:
+    - requests: Library for making HTTP requests.
+
+Note: Make sure to install the 'requests' library before using this module.
+You can install it by running 'pip install requests' in your Python environment.
```

Comment:
I love using ChatGPT for documentation too, some of its output is irrelevant (we use potery for that case)

Reply:
removed suggestion to install using pip.